docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+(extending_client_side)=
+
+# Extending client-side behaviour
+
+Many kinds of common customisations can be done without reaching into JavaScript, but depending on what parts of the client-side interaction you want to leverage or customise, you may need to employ React, Stimulus, or vanilla (plain) JS.
+
+[React](https://reactjs.org/) is used for more complex parts of Wagtail, such as the sidebar, commenting system, and the Draftail rich-text editor.
+For basic JavaScript-driven interaction, Wagtail is migrating towards [Stimulus](https://stimulus.hotwired.dev/).
+
+You don't need to know or use these libraries to add your custom behaviour to elements, and in many cases vanilla (plain) JavaScript will work fine, but Stimulus is the recommended approach for anything that vanilla JavaScript can't do.
+
+You don't need to have Node.js tooling running for your custom Wagtail installation for many customisations built on these libraries, but in some cases, such as building packages, it may make more complex development easier.
+
+```{note}
+it's recommended that you avoid using jQuery as this will be removed in a future version of Wagtail.
+```
+
+(extending_client_side_injecting_javascript)=
+
+## Adding custom JavaScript
+
+Within Wagtail's admin interface, there are a few ways to add JavaScript.
+
+The simplest way is to add global JavaScript files via hooks, see [](insert_editor_js) and [](insert_global_admin_js).
+
+For JavaScript added when a specific Widget is used you can add an inner `Media` class to ensure that the file is loaded when the widget is used, see [Django's docs on their form `Media` class](https://docs.djangoproject.com/en/stable/topics/forms/media/#assets-as-a-static-definition).
+
+In a similar way, Wagtail's [template components](template_components) provide a `media` property or `Media` class to add scripts when rendered.
+
+These will ensure the added files are used in the admin after the core JavaScript admin files are already loaded.
+
+(extending_client_side_using_events)=
+
+## Extending with DOM events
+
+When approaching client-side customisations or adopting new components, try to keep the implementation simple first, you may not need any knowledge of Stimulus, React, JavaScript Modules or a build system to achieve your goals.
+
+The simplest way to attach behaviour to the browser is via DOM Events and plain (vanilla) JavaScript.
+
+For example, if you want to attach some logic to a field value change in Wagtail you can add an event listener, check if it's the correct element and change what you need.
+
+```javascript
+// myapp/static/js/events.js
+document.addEventListener('change', function (event) {
+    if (event.target) {
```

Comment:
This feels dated to me to add global event listeners to interact with a single field. Is this so people can implement those changes without worrying about the execution order of the JS initialisation?

Reply:
The idea is that JS integrations do not need to be complex and event listeners are an elegant, simple, maintainable way to get many kinds of customisations working. 

Additionally, it paves the way for Stimulus usage which is built on Events as first class citizens.

I'd also push back a bit on event listeners being dated, I think it's a very useful way to add behaviour that the browser is built around.

Reply:
I think Thibaud's point is about global event listeners, not event listeners in general. Adding global event listeners to specific elements should be fine, but adding event listeners to the document element means that the function is executed even when the event comes from an element that is not of the function's concerns (which I think is what Thibaud meant by "wasteful" in another comment).

There are cases where this is useful, e.g. capturing a "click away", but in most cases I don't think it's a good idea to advise this.

Reply:
Thanks both, I will just remove this example and keep this section short and generic.

The underlying goal is to remind developers that they may not need to reach for a complex solution when something simple may get the job done.